packages/json5/json5.mjs

```diff
@@ -0,0 +1,52 @@
+import { getFilenameExt } from '@nodejs-loaders/parse-filename';
+import JSON5 from 'json5';
+
+/** @typedef {import('esbuild').TransformOptions} ESBuildOptions */
+/** @typedef {`file://${string}`} FileURL */
+
+/**
+ * @type {import('node:module').ResolveHook}
+ */
+async function resolveJSON5(specifier, ctx, nextResolve) {
+	const resolved = await nextResolve(specifier);
+
+	if (getFilenameExt(specifier) !== '.json5') {
+		return resolved;
+	}
+
+	return { ...resolved, format: 'json5' };
+}
+export { resolveJSON5 as resolve };
+
+/**
+ * @type {import('node:module').LoadHook}
+ * @argument {FileURL} url
```

Comment:
```suggestion
```

Reply:
in this loader it's didn't create any issue so let's keep it simple (I check on jsonc loader)

Reply:
Def nit, but FileUrl is more robust. `string` is a cancer of a type in node.